compact/ecal_barrel_interlayers.xml

```diff
@@ -12,7 +12,17 @@
       To change the number of imaging layers from, e.g., 9 to 6,
       change EcalBarrelImagingLayers_nMax variable
     </comment>
-    <constant name="EcalBarrelImagingLayers_nMax"  value="6"/>
+    <constant name="EcalBarrelImagingLayers_nMax"   value="6"/>
+    <constant name="EcalBarrel_Calorimeter_zmin"    
+      value="min(257*cm, EcalBarrelBackward_zmax)"/>
+    <constant name="EcalBarrel_Calorimeter_zmax"    
+      value="min(177.5*cm, EcalBarrelForward_zmax)"/>
+    <constant name="EcalBarrel_Readout_zmin"        value="272*cm"/>
+    <constant name="EcalBarrel_Readout_zmax"        value="192.5*cm"/>
+    <constant name="EcalBarrel_Calorimeter_length"
+      value="EcalBarrel_Calorimeter_zmax + EcalBarrel_Calorimeter_zmin"/>
+    <constant name="EcalBarrel_Calorimeter_offset"
+      value="(EcalBarrel_Calorimeter_zmin - EcalBarrel_Calorimeter_zmax)/2.0"/>
```

Comment:
So, now here, the offset (which is the average of forward and backward signed positions) should be
```suggestion
    <constant name="EcalBarrel_Calorimeter_offset"
      value="(EcalBarrel_Calorimeter_zmax - EcalBarrel_Calorimeter_zmin)/2.0"/>
```